### ISSUE TITLE
ConvertToDiffractionMDWorkspace failing with input in dSpacing

### ISSUE THREAD

**Issue by DannyHindson** [mantidproject/mantid#31685]
Issue reported by Pascal Manuel

The ConvertToDifractionMDWorkspace algorithm fails when provided with an input workspace in dSpacing with this error:

![image](https://user-images.githubusercontent.com/56295817/120453640-21362880-c38b-11eb-835b-49d12c49b189.png)

**Comment by RichardWaiteSTFC** [comment]
FYI bit quicker to test if you crop workspace!
```
Load(Filename='SXD23767.raw', OutputWorkspace='SXD23767', LoadMonitors='Exclude')
CropWorkspace(InputWorkspace='SXD23767', OutputWorkspace='SXD23767', XMin=6000, XMax=9000) 
ConvertUnits(InputWorkspace='SXD23767', OutputWorkspace='SXD23767', Target='dSpacing')
wsMD = ConvertToDiffractionMDWorkspace(InputWorkspace='SXD23767', OutputDimensions= 'Q (lab frame)')
```

**Comment by DannyHindson** [comment]
Seems to be a problem in ConvertToMDMinMaxLocal which is called as a child algorithm from ConvertToDiffractionMDWorkspace. Any spectra that are monitors or don't have detectors aren't being handled properly.

**Comment by martyngigg** [comment]
Fixed by #31693

**Comment by DannyHindson** [comment]
Few comments on the existing test coverage here.
Firstly, in common with other algorithms it's a bit arbitrary as to whether the tests go into a system test or a unit test.
The current unit tests for ConvertToDiffractionMDWorkspace all use a "mini TOPAZ" instrument definition and they all assume the input workspace has units of TOF. So probably written by SNS and focussed on their typical usage. The simplest thing to plug this specific bug would be to add a further unit test that has an input workspace with units of dSpacing. I'll create an issue for this.
Separate to this we have scheduled a task in the 6.2 S1 diffraction sprint to look at improving the test coverage for the WISH single crystal workflow (#31009). This will probably take the form of a system test.
